src/Symfony/Bundle/TwigBundle/TwigEngine.php

```diff
@@ -74,8 +74,12 @@ public function render($name, array $parameters = array())
             if ($name instanceof TemplateReference) {
                 try {
                     // try to get the real name of the template where the error occurred
-                    $name = $e->getTemplateName();
-                    $path = (string) $this->locator->locate($this->parser->parse($name));
+                    if (method_exists($e, 'getSourceContext')) {
+                        $templateName = $e->getSourceContext()->getName();
+                    } else {
+                        $templateName = $e->getTemplateName();
+                    }
+                    $path = (string) $this->locator->locate($this->parser->parse($templateName));
```

Comment:
But $name is a parameter that is already in use, and we even checking if ```$name instanceof TemplateReference```. Should we overwrite this variable?